src/Scripting/CSharpTest/ObjectFormatterTests.cs

```diff
@@ -437,23 +439,57 @@ public void DebuggerProxy_FrameworkTypes_MdArray()
         }
 
         [Fact]
-        public void DebuggerProxy_FrameworkTypes_IEnumerable()
+        public void DebuggerProxy_FrameworkTypes_IEnumerable_Core()
         {
             string str;
             object obj;
 
-            obj = Enumerable.Range(0, 10);
+            obj = Range_Core(0, 10);
             str = s_formatter.FormatObject(obj, SingleLineOptions);
 
-            // the implementation differs between .NET Core and .NET FX
-            if (str.StartsWith("Enumerable"))
-            {
-                Assert.Equal("Enumerable.RangeIterator(Count = 10)", str);
-            }
-            else
-            {
-                Assert.Equal("RangeIterator { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 }", str);
-            }
+            Assert.Equal("ObjectFormatterTests.CoreRangeIterator(Count = 10)", str);
+        }
+
+        // This method and the class below emulate the behaviour of Enumerable.Range
+        // in .NET Core
```

Comment:
```suggestion
        // This method and the class below emulate the behaviour of Enumerable.Range
        // in .NET Core. We use a custom type since not all runtime implementations (e.g. Mono)
        // apply precisely the same attributes, but we want to test behavior under a specific set
        // of attributes.
```